### DISCUSSION TITLE
Any way to programmatically call a layout?

### DISCUSSION THREAD

**Discussion by hartwm** [sveltejs/kit#5599] [Q&A]
hardcoding a file to include @foo.svelte is not always possible when you are dynamically generating routes via [slug].svelte etc. Is there a way to call layouts in load based on CMS data or something like that?

**Comment by mustofa-id** [comment]
No, but you can always define a layout per route. If you want something similar to Gatsby, which allows you to select a layout in the post page, you can integrate Kit with [mdsvex](https://github.com/pngwn/mdsvex) and use its [layout](https://mdsvex.com/docs#layouts).

**Comment by xpertekShaun** [comment]
__layout.svelte files are just svelte components that take a main slot.

you can import any layout file as you would a regular component file

```
<script>
    import Layout1 from '../edit/[endpoint]/[id]/__layout.svelte'
    import Layout2 from '../delete/[endpoint]/[id]/__layout.svelte'
    import { page } from '$app/stores'
    let { slug } = $page.params
    let Layout = slug === 'special' ? Layout1 : Layout2
</script>

<svelte:component this={Layout}>Im inside that layout file</svelte:component>

```

You get no benefits though of route inheritance, so in that case it will be better to setup specific routes for special [slug]'s

**Comment by nickbreaton** [comment]
On mobile so haven't been able to test this, but you could possibly use param matchers to determine which layout matches for that slug. 

[slug=foo]@foo.svelte
[slug=bar]@bar.svelte

https://kit.svelte.dev/docs/routing#advanced-routing-matching

**Reply by nickbreaton** [reply]
You might run into a sequential request issue if Kit queries them one after another.
